### PR TITLE
Add fuction to run shell commands

### PR DESCRIPTION
**Pull request by JonParkes**
Adds a function to slave that will run any commad that shell will run, wait till completion, and output stdout/stderr into a file named result.txt

issue #4

### CODE REVIEW

File: src/slave/slave.py [hunk 1]
```diff
@@ -124,6 +126,15 @@ def start(self):
 			sleep(1)
 		self.handle_job()
 
+	def run_cpp(command):
```
Comment (ansh-projects) [inline src/slave/slave.py:129]: Probably should name function after what it does, which is run shell command.

File: src/slave/slave.py [hunk 2]
```diff
@@ -124,6 +126,15 @@ def start(self):
...
+		"""
+		Execute a c++ executable and create a result file
+		"""
+		args = shlex.split(command)
+
+		with open('result.txt', "w") as f:
+			run(args, stdout=f, stderr=f, text=True)
```
Comment (ansh-projects) [inline src/slave/slave.py:136]: run returns a CompletedProcess object. Should check it's return code to see if it was successful or not, and then handle it

File: src/slave/slave.py [hunk 3]
```diff
@@ -124,6 +126,15 @@ def start(self):
 			sleep(1)
 		self.handle_job()
 
+	def run_cpp(command):
+		"""
+		Execute a c++ executable and create a result file
+		"""
+		args = shlex.split(command)
+
+		with open('result.txt', "w") as f:
```
Comment (ansh-projects) [inline src/slave/slave.py:135]: Wouldn't mind a more descriptive log file name, but not important

File: src/slave/slave.py (hunk 1 above)
Comment (tbgiles) [reply src/slave/slave.py:129]: I second this

Comment (JonParkes) [reply src/slave/slave.py:129]: oops i changed this but i guess my git lost the changes somehwere, thats my bad

File: src/slave/slave.py (hunk 3 above)
Comment (JonParkes) [reply src/slave/slave.py:135]: do you have any suggestions? My assumtion was there would be one command run so the file would be the results of that command

File: src/slave/slave.py (hunk 2 above)
Comment (JonParkes) [reply src/slave/slave.py:136]: ya you are right i overlooked this when i wrote it

File: src/slave/slave.py (hunk 3 above)
Comment (ansh-projects) [reply src/slave/slave.py:135]: ApplicationResultLog? Idk

Comment (JonParkes) [reply src/slave/slave.py:136]: sure that works

File: src/slave/slave.py (hunk 2 above)
Comment (JonParkes) [reply src/slave/slave.py]: I have made it so it returns the returncode of the command and then the place using the function will handle it based on what command it used

**Pull request merged**